src/utils.js

```diff
@@ -1,17 +1,31 @@
-export function findScript(url) {
-    return document.querySelector(`script[src="${url}"]`);
+export function findScript(url, dataAttributes) {
+    const currentScript = document.querySelector(`script[src="${url}"]`);
+    if (!currentScript) return null;
+
+    const nextScript = createScriptElement(url, dataAttributes);
+
+    // check if the new script has the same number of data attributes
+    if (objectSize(currentScript.dataset) !== objectSize(nextScript.dataset)) {
```

Comment:
@mnicpt thanks for pairing on this. I added the `objectSize` helper function and some comments to improve readability while still keeping it ES3. Please give this another review.